test/Libraries/WorkflowTests/GeometryDefectTests.cs

```diff
@@ -504,6 +504,59 @@ public void TestCoordinateSystem()
             Assert.AreEqual(0,csPreview.Origin.X );
             Assert.AreEqual(0,csPreview.Origin.Y);
             Assert.AreEqual(0, csPreview.Origin.Z); 
-        }       
+        }
+
+
+        [Test]
+        public void VoronoiByParameterOnSurface_MAGN_8039()
+        {
+            // http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8039
+
+            string openPath = Path.Combine(TestDirectory,
+            @"core\WorkflowTestFiles\GeometryDefects\Voronoi.ByParameterOnSurface_MAGN_8039.dyn");
+
+            RunModel(openPath);
+
+            AssertNoDummyNodes();
+
+            // check all the nodes and connectors are loaded
+            Assert.AreEqual(9, CurrentDynamoModel.CurrentWorkspace.Nodes.Count());
+            Assert.AreEqual(11, CurrentDynamoModel.CurrentWorkspace.Connectors.Count());
+
+            var VoronoiNodeID = "2fdd3ad8-dc1e-4087-be18-196732d41d7a";
+
+            for (int i = 0; i <= 10; i++)
+            {
+                var allLines = GetPreviewValueAtIndex(VoronoiNodeID, i) as Line;
+                Assert.IsNotNull(allLines);
```

Comment:
Actually I kept it random and not checking for the total number of lines as well. I doubt I can do the verification based on any particular line. The main issue was it was throwing unhandled exception, which will definitely caught by this test case.